product_configurator/models/product.py

```diff
@@ -269,6 +269,32 @@ def get_config_image_obj(self, value_ids, size=None):
         return img_obj
 
     @api.multi
+    def encode_custom_values(self, custom_values):
+        """ Hook to alter the values of the custom values before creating or writing
+
+            :param custom_values: dict {product.attribute.id: custom_value}
+
+            :returns: list of custom values compatible with write and create
+         """
+        custom_values = custom_values or {}
```

Comment:
Hey, I can understand from a purist point of view, so am happy to make a change.  In fact, it should only be called with a dictionary, so will take this out...